src/sidebar/directive/annotation-share-dialog.js

```diff
@@ -4,73 +4,69 @@ var angular = require('angular');
 
 var scopeTimeout = require('../util/scope-timeout');
 
-module.exports = function () {
-  return {
-    bindToController: true,
-    controllerAs: 'vm',
-    restrict: 'E',
-    template: require('../templates/annotation_share_dialog.html'),
-    // @ngInject
-    controller: function ($scope, $element) {
-      var shareLinkInput = $element.find('input')[0];
+module.exports = {
+  controllerAs: 'vm',
+  template: require('../templates/annotation_share_dialog.html'),
+  // @ngInject
+  controller: function ($scope, $element) {
+    var shareLinkInput = $element.find('input')[0];
 
-      $scope.$watch('vm.isOpen', function (isOpen) {
-        if (isOpen) {
-          // Focus the input and select it once the dialog has become visible
-          scopeTimeout($scope, function () {
-            shareLinkInput.focus();
-            shareLinkInput.select();
-          });
+    $scope.$watch('vm.isOpen', function (isOpen) {
```

Comment:
Do you think follow-up work could replace all instances of `$watch` with `$onChanges` and/or `$doCheck`?